pytorch_lightning/trainer/seed.py

```diff
@@ -0,0 +1,29 @@
+"""Helper functions to help with reproducibility of models """
+import os
+
+import numpy as np
+import random
+import torch
+
+from pytorch_lightning import _logger as log
+
+
+def seed_everything(seed: int = None):
+    """Function that sets seed for pseudo-random number generators  in:
+        pytorch, numpy, python.random and sets PYTHONHASHSEED environment variable.
+    """
+    max_seed_value = np.iinfo(np.uint32).max
+    min_seed_value = np.iinfo(np.uint32).min
+
+    try:
+        seed = int(seed)
+    except (TypeError, ValueError):
+        seed = random.randint(0, max_seed_value)
+        log.info(f"No seed found, seed set to {seed}")
+
+    assert seed <= max_seed_value, "seed is too big, numpy accepts only uint32"
+    assert seed >= min_seed_value, "seed is too small, numpy accepts only uint32"
+    os.environ["PYTHONHASHSEED"] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
```

Comment:
I'm not exactly an expert so maybe the function _seed_everything_ is already performing as expected (or this problem is already solved differently) but I noticed that I did not obtained reproducibility without also adding: 

`torch.backends.cudnn.deterministic = True`

Reply:
This can be done by setting the `benchmark` flag of the trainer. And We did not include this on purpose, since this hasn't anything to do with seeding, but specifies how to choose the algorithms used by cudnn

Reply:
I agree with @justusschock :)

Reply:
Also, I added `deterministic` in the trainer parameters

Reply:
Also, I added parameter `deterministic` to the trainer class.
And added warning in the trainer: 
```
        # seeding random number generators
        if (seed is not None) and not deterministic:
            log.warning("To ensure full reproducibility deterministic flag should also be set")
        seed_everything(seed)
        self.deterministic = deterministic
        torch.backends.cudnn.deterministic = self.deterministic
```

Reply:
Yes and I kindly ask you to remove it again, since we already have an argument for that: `benchmark`

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/1572#discussion_r416518420